examples/40_advanced/example_text_preprocessing.py

```diff
@@ -22,67 +22,54 @@
-
-# load train data
-df_train = pd.DataFrame({"X": [], "y": []})
-
-for idx, (text, target) in enumerate(
-    zip(newsgroups_train.data, newsgroups_train.target)
-):
-    df_train = pd.concat(
-        [
-            df_train,
-            pd.DataFrame(
-                {"X": text, "y": newsgroups_train.target_names[target]}, index=[idx]
-            ),
-        ]
-    )
-
-# explicitly label text column as string
-X_train = df_train.astype({"X": "string", "y": "category"})
-
-# show all 20 labels
-print(list(newsgroups_train.target_names))
-
-# reduce the example to only 5 labels
-five_newsgroups_labels = list(newsgroups_train.target_names)[:5]
-
-X_train = X_train[~X_train["y"].isin(five_newsgroups_labels)]
-y_train = X_train.pop("y")
-
-# load test data
-df_test = pd.DataFrame({"X": [], "y": []})
-
-for idx, (text, target) in enumerate(zip(newsgroups_test.data, newsgroups_test.target)):
-    df_test = pd.concat(
-        [
-            df_train,
-            pd.DataFrame(
-                {"X": text, "y": newsgroups_train.target_names[int(target)]},
-                index=[idx],
-            ),
-        ]
-    )
-
-# explicitly label text column as string
-X_test = df_test.astype({"X": "string", "y": "category"})
-X_test = X_test[~X_test["y"].isin(five_newsgroups_labels)]
-y_test = X_test.pop("y")
+############################################################################
+# Creating a pandas dataframe
+# ===========================
+# Both categorical and text features are often strings. Python Pandas stores python stings
+# in the generic `object` type. Please ensure that the correct
+# `dtype <https://pandas.pydata.org/docs/user_guide/basics.html#dtypes>` is applied to the correct
+# column.
+
+# create a pandas dataframe for training labeling the "Text" column as sting
+X_train = pd.DataFrame({"Text": pd.Series(X_train, dtype="string")})
+
+# create a pandas dataframe for testing labeling the "Text" column as sting
+X_test = pd.DataFrame({"Text": pd.Series(X_test, dtype="string")})
 
 ############################################################################
 # Build and fit a classifier
 # ==========================
 
+# create an autosklearn Classifier or Regressor depending on your task at hand.
 automl = autosklearn.classification.AutoSklearnClassifier(
-    # set the time high enough text preprocessing can create many new features
-    time_left_for_this_task=300,
-    per_run_time_limit=30,
+    time_left_for_this_task=60,  # absolute time limit for fitting the ensemble
+    per_run_time_limit=30,  # time limit for single models (ensures seeing a variety of models)
     tmp_folder="/tmp/autosklearn_text_example_tmp",
 )
-automl.fit(X_train, y_train, dataset_name="20_Newsgroups")
+
+automl.fit(  # fit our model to the training data
+    X=X_train,  # passing training data (encoded as pandas dataframe)
+    y=y_train,  # passing training labels
+    # ('array like' object: pandas Series, numpy array, python list etc.)
```

Comment:
I don't fully understand why this comment is here.

Reply:
i will minimize it